Add mysql bigip monitor module

andreykashcheev opened the pull request:
### Summary 

This change is intended to add a new module intended to configure BIGIP Monitor for MySQL. 
Internal ticket: FMFA-431

### Changes

  - Add bigip_monitor_mysql module
  - Add unit tests for new bigip_monitor_module
  - Add integration test for new bigip_monitor_module

### Additional Information

WIP status will be resolved when MR will be ready to be approved and merged

wojtek0806 commented:
full tests against v12 - v15 and post the results this is before i can review this

andreykashcheev commented:
I have run full tests on all versions (12,13,14 and 15); please see tests outputs below:

PLAY [Test the bigip_monitor_mysql module] ******************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] **************************************************************************************************************************************************************************************************************************************************
ok: [bigip_v13]
ok: [bigip_v14]
ok: [bigip_v15]
ok: [bigip_v12]

TASK [bigip_monitor_mysql : Create MySQL Monitor with description] ******************************************************************************************************************************************************************************************************
changed: [bigip_v12]
changed: [bigip_v14]
changed: [bigip_v15]
changed: [bigip_v13]

TASK [bigip_monitor_mysql : Assert Create MySQL Monitor with description] ***********************************************************************************************************************************************************************************************
ok: [bigip_v15] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v14] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v13] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v12] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_monitor_mysql : Create MySQL Monitor with description - Idempotent check] ***********************************************************************************************************************************************************************************
ok: [bigip_v12]
ok: [bigip_v14]
ok: [bigip_v13]
ok: [bigip_v15]

TASK [bigip_monitor_mysql : Assert Create MySQL Monitor with description - Idempotent check] ****************************************************************************************************************************************************************************
ok: [bigip_v15] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v14] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v13] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v12] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_monitor_mysql : Modify MySQL Monitor] ***********************************************************************************************************************************************************************************************************************
changed: [bigip_v12]
changed: [bigip_v13]
changed: [bigip_v14]
changed: [bigip_v15]

TASK [bigip_monitor_mysql : Assert Modify MySQL Monitor] ****************************************************************************************************************************************************************************************************************
ok: [bigip_v15] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v14] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v13] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v12] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_monitor_mysql : Modify MySQL Monitor - Idempotent check] ****************************************************************************************************************************************************************************************************
ok: [bigip_v12]
ok: [bigip_v14]
ok: [bigip_v13]
ok: [bigip_v15]

TASK [bigip_monitor_mysql : Assert Change MySQL Monitor with description  - Idempotent check] ***************************************************************************************************************************************************************************
ok: [bigip_v15] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v14] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v13] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v12] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_monitor_mysql : Remove MySQL Monitor description] ***********************************************************************************************************************************************************************************************************
changed: [bigip_v12]
changed: [bigip_v13]
changed: [bigip_v15]
changed: [bigip_v14]

TASK [bigip_monitor_mysql : Assert Remove MySQL Monitor description] ****************************************************************************************************************************************************************************************************
ok: [bigip_v15] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v14] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v13] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v12] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_monitor_mysql : Remove MySQL Monitor description - Idempotent check] ****************************************************************************************************************************************************************************************
ok: [bigip_v12]
ok: [bigip_v14]
ok: [bigip_v13]
ok: [bigip_v15]

TASK [bigip_monitor_mysql : Assert Remove MySQL Monitor description - Idempotent check] *********************************************************************************************************************************************************************************
ok: [bigip_v15] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v14] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v13] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [bigip_v12] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP **************************************************************************************************************************************************************************************************************************************************************
bigip_v12                  : ok=13   changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
bigip_v13                  : ok=13   changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
bigip_v14                  : ok=13   changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
bigip_v15                  : ok=13   changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

andreykashcheev commented on library/modules/bigip_monitor_mysql.py:
```diff
@@ -0,0 +1,928 @@
+        special characters, the system may require that the string be enclosed with single quotation marks.
+      - If this value is null, then a valid connection suffices to determine that the service is up.
+        In this case, the system does not need the recv, recv-row, and
+        recv-column options and ignores the options even if not null.
```
This commend is addressed.

andreykashcheev commented on library/modules/bigip_monitor_mysql.py:
```diff
@@ -0,0 +1,928 @@
+options:
+  name:
+    description:
+      - Monitor name .
```
This commend is addressed.
